sharktuner/sharktuner/candidate_ordering.py

```diff
@@ -104,3 +108,95 @@ def reorder_assignments(
             return indices
         case _:
             assert False
+
+
+@dataclass
+class TuningRecord:
+    gen_id: int  # Original index from candidate generation.
+    candidate_id: int  # Index in candidate_trackers after sorting.
+    knob: Optional[common.KnobAssignment] = None
+    to_compile: bool = False
+    compile_status: bool = False
+    to_benchmark: bool = False
+    benchmark_device_id: Optional[str] = None
+    benchmark_queue_position: Optional[int] = None
+    benchmark_status: bool = False
+    baseline_benchmark_time_us: Optional[float] = None
+    benchmark_time_us: Optional[float] = None
+    benchmark_speedup: Optional[float] = None
+    benchmark_rank_order: Optional[int] = None
+
+
+def init_tuning_records(
+    knobs: list[Optional[common.KnobAssignment]], sorted_order: list[int]
+) -> list[TuningRecord]:
+    tuning_records: list[TuningRecord] = []
+    tuning_records.append(
+        TuningRecord(gen_id=0, candidate_id=0, to_compile=True, to_benchmark=True)
+    )
+
+    for can_idx, gen_idx in enumerate(sorted_order, start=1):
```

Comment:
can you expand these variable names? I think `can_index` is the actual index while `gen_idx` is the original order form z3?